api/subscriptions/types_test.go

```diff
@@ -147,7 +148,8 @@ func TestConvertTransfer(t *testing.T) {
 func TestConvertEventWithBadSignature(t *testing.T) {
 	// Arrange
 	var sig [65]byte
-	rand.Read(sig[:])
+	_, err := rand.Read(sig[:])
```

Comment:
I think we need to set here too the last byte to a value greater than 4, if it's random we could end up in the same failing scenario

Reply:
Maybe put a for loop 1000 times and see if it fails?

Reply:
@paologalligit oops, I missed this one, let me do a quick refactor.

@darrenvechain  if you add `sig[64] = 1` it always fails the test